mesh/meshdb.go

```diff
@@ -275,43 +275,40 @@ func (m *DB) ForBlockInView(view map[types.BlockID]struct{}, layer types.LayerID
 
 // LayerBlockIds retrieves all block ids from a layer by layer index
 func (m *DB) LayerBlockIds(index types.LayerID) ([]types.BlockID, error) {
-	idsBytes, err := m.layers.Get(index.Bytes())
-	if err != nil {
-		return nil, err
-	}
-
-	if len(idsBytes) == 0 {
-		// zero block layer
-		return []types.BlockID{}, nil
+	layerBuf := index.Bytes()
+	zero := false
+	ids := []types.BlockID{}
+	it := m.layers.Find(layerBuf)
+	// TODO(dshulyak) iterator must be able to return an error
+	for it.Next() {
+		if len(it.Key()) == len(layerBuf) {
+			zero = true
+			continue
```

Comment:
should this be `break` instead? i guess if we have late blocks for this layer we don't clear the zero block entry.

Reply:
:+1:

Reply:
the TestMesh_SetZeroBlockLayer started to fail. i think i will keep continue to preserve original behavior for now